new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleTraefikValidation.java

```diff
@@ -0,0 +1,209 @@
+
+    // get a new unique traefik Namespace
+    tfNamespace = createNamespace();
+    logger.info(String.format("Created a new namespace called %s", tfNamespace));
+
+    String projectRoot = System.getProperty("user.dir");
+
+    // helm install parameters
+    tfHelmParams = new HelmParams()
+        .releaseName("traefik-operator")
+        .namespace(tfNamespace)
+        .repoUrl("https://kubernetes-charts.storage.googleapis.com/")
+        .repoName("stable")
+        .chartName("traefik")
+        .chartValuesFile(projectRoot + "/../kubernetes/samples/charts/traefik/values.yaml");
+
+    TraefikParams tfParams = new TraefikParams()
+        .helmParams(tfHelmParams)
+        .nameSpaces("{" + tfNamespace + "}");
+
+    // install Traefik
+    assertThat(installTraefik(tfParams))
+        .as("Test installTraefik returns true")
+        .withFailMessage("installTraefik() did not return true")
+        .isTrue();
+    logger.info(String.format("Traefik is installed in namespace %s", tfNamespace));
+
+    assertThat(helmList(tfHelmParams))
+        .as("Test helmList returns true")
+        .withFailMessage("helmList() did not return true")
+        .isTrue();
+
+    // waiting for an async operation to complete.
+    // we first wait 30 seconds, then we check if the traefik pod is running.
+    with().pollDelay(30, SECONDS)
+        // we check again every 10 seconds.
+        .and().with().pollInterval(10, SECONDS)
+        // this listener lets us report some status with each poll
+        .conditionEvaluationListener(
+            condition -> logger.info(
+                "Waiting for traefik to be running (elapsed time {0}ms, remaining time {1}ms)",
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        // and here we can set the maximum time we are prepared to wait
+        .await().atMost(5, MINUTES)
+        .until(traefikIsRunning(tfNamespace));
+  }
+
+  @Test
+  @Order(2)
+  @DisplayName("Upgrade the Traefik release")
+  public void testUpgradingTraefik() {
+
+    // create a new unique namespace for wls domain
+    domainNamespace1 = createNamespace();
+    logger.info(String.format("Created a new namespace called %s", domainNamespace1));
+
+    // helm install parameters
+    tfHelmParams = new HelmParams()
+            .releaseName("traefik-operator")
+            .namespace(tfNamespace)
+            .chartDir("stable/traefik");
+
+    TraefikParams tfParams = new TraefikParams()
+            .helmParams(tfHelmParams)
+            .nameSpaces("{" + tfNamespace + "," + domainNamespace1 + "}");
+
+    // upgrade Traefik
+    assertThat(upgradeTraefik(tfParams))
+            .as("Test upgradeTraefik returns true")
+            .withFailMessage("upgradeTraefik() did not return true")
+            .isTrue();
+    logger.info(String.format("Traefik is upgraded in namespace %s", tfNamespace));
+
+    assertThat(helmGetExpectedValues(tfHelmParams, domainNamespace1))
```

Comment:
This looks like an assert of an assert.  I'd rather see a method to do the helm list (or whatever call this is) and then assert here that the values include what you need.

Reply:
added a generic method to verify the result.